Fedora CI build broken since 2025-12-04: `mesa-libOSMesa-devel not found`

Hi, I've been using this project for a while, but recently when I tried to use the latest version, I noticed the Fedora CI build has been broken since Dec 4, 2025.

Because I'd like to test this based on the latest Wine master on another personal project, I'll try to dig into this.

Error log shows:
```log
...
package mesa-libOSMesa-devel is not installed
=> Installing package: mesa-libOSMesa-devel | Using dnf
Updating and loading repositories:
Repositories loaded.
Failed to resolve the transaction:
No match for argument: mesa-libOSMesa-devel
You can try to add to command line:
--skip-unavailable to skip unavailable packages
==> WARNING: Failed to install package: mesa-libOSMesa-devel
package mesa-libOSMesa-devel is not installed
=> Installing package: mesa-libOSMesa-devel | Using dnf| Attempt: 2/3
Updating and loading repositories:
Repositories loaded.
Failed to resolve the transaction:
No match for argument: mesa-libOSMesa-devel
You can try to add to command line:
--skip-unavailable to skip unavailable packages
==> WARNING: Failed to install package: mesa-libOSMesa-devel
package mesa-libOSMesa-devel is not installed
=> Installing package: mesa-libOSMesa-devel | Using dnf| Attempt: 3/3
Updating and loading repositories:
Repositories loaded.
Failed to resolve the transaction:
No match for argument: mesa-libOSMesa-devel
You can try to add to command line:
--skip-unavailable to skip unavailable packages
==> WARNING: Failed to install package: mesa-libOSMesa-devel
===> ERROR: Failed to install package: mesa-libOSMesa-devel after 3 attempts.
===> ERROR: 64-bit dependencies installation failed. Please check the error message and install the missing dependencies manually.
=> Removed BIG_UGLY_FROGMINER - Ribbit
=> Removed Proton-tkg token - Valve Ribbit
=> exit cleanup done
Error: Process completed with exit code 1.
```

The error occurs because `mesa-libOSMesa-devel` package doesn't exist in Fedora 43, but none of the dependencies nor the Fedora build scripts were changed that day. After looking for the root cause, turns out that the CI base image was not pinned and that on the same day the CI started failing, the official Fedora Docker image switched `fedora:latest` from Fedora 42 to Fedora 43 (see [this commit](https://github.com/docker-library/repo-info/commit/4ca9c60f058a0488ff778809acdec583be7b5bfb) from them).

Since there's too many things to adjust because Fedora 43 deprecated YASM, and I've found a lot of potential compatibility issues that requires tinkering, I'll just try pinning the image to `fedora:42`.

--- a/.github/workflows/wine-fedora.yml
+++ b/.github/workflows/wine-fedora.yml
@@ -8,7 +8,7 @@ on:
 jobs:
   build:
     runs-on: ubuntu-latest
-    container: fedora:latest
+    container: fedora:42
     steps:
 
       - uses: actions/checkout@v4

